game-core/src/main/java/games/strategy/triplea/delegate/power/calculator/AaDefenseCombatValue.java

```diff
@@ -0,0 +1,95 @@
+package games.strategy.triplea.delegate.power.calculator;
+
+import games.strategy.engine.data.GameData;
+import games.strategy.engine.data.Unit;
+import games.strategy.triplea.attachments.UnitSupportAttachment;
+import java.util.function.Predicate;
+import lombok.AccessLevel;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.Value;
+
+/**
+ * Calculates defense strength and roll for AA/Targeted dice
+ *
+ * <p>This takes into account friendly support, and enemy support
+ */
+@Builder
+@Value
+@Getter(AccessLevel.NONE)
+class AaDefenseCombatValue implements CombatValue {
+
+  @NonNull GameData data;
+  @NonNull AvailableSupports friendlySupportTracker;
+  @NonNull AvailableSupports enemySupportTracker;
+
+  @Override
+  public StrengthOrRollCalculator getRoll() {
+    return new AaDefenseRoll(friendlySupportTracker, enemySupportTracker);
+  }
+
+  @Override
+  public StrengthOrRollCalculator getStrength() {
+    return new AaDefenseStrength(data, friendlySupportTracker, enemySupportTracker);
+  }
+
+  @Override
+  public boolean isDefending() {
+    return true;
+  }
+
+  @Override
+  public GameData getGameData() {
+    return data;
```

Comment:
nit, any thoughts to rename `data` to `gameData`, then you can use a lombok `@Getter` to perhaps clean up the code a little bit.